From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Haotien Hsu <haotienh@nvidia.com>
Date: Tue, 6 Jan 2026 08:13:10 +0000
Subject: [PATCH 228/241] NVIDIA: SAUCE: PCI/DOE: Increase DOE timeout

When we establish the PCIe Link IDE, we need the PSC to perform the key
management (KM) process over DOE on our platforms. The PSC processing
time for KM on the endpoint (EP) side may be up to 10 seconds, so we
increased the DOE timeout to 16 seconds for Link IDE establishment.

This is a temporary patch, and we will revert it when the PSCFW latency
is fixed.

Bug 5800640
Jira NET-4222

Change-Id: Ibdbd8d5ef56e378c00c9b874dcdf03754b8913c0
Signed-off-by: Haotien Hsu <haotienh@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3521666
Reviewed-by: WK Tsai <wtsai@nvidia.com>
Reviewed-by: Manikanta Maddireddy <mmaddireddy@nvidia.com>
(cherry picked from commit e9a584694b502653b5e024fd5f0043dad13acaab)

diff --git a/drivers/pci/doe.c b/drivers/pci/doe.c
index aae9a8a00406e..3d521f4b0b349 100644
--- a/drivers/pci/doe.c
+++ b/drivers/pci/doe.c
@@ -27,7 +27,7 @@
 #define PCI_DOE_FEATURE_DISCOVERY 0
 
 /* Timeout of 1 second from 6.30.2 Operation, PCI Spec r6.0 */
-#define PCI_DOE_TIMEOUT HZ
+#define PCI_DOE_TIMEOUT ((HZ) << 4)
 #define PCI_DOE_POLL_INTERVAL	(PCI_DOE_TIMEOUT / 128)
 
 #define PCI_DOE_FLAG_CANCEL	0
-- 
2.43.0

